build: automate build & launch of storybook

andreaslordos opened the pull request:
🏆 Enhancements
- Added `build.py` in the `/packages/superset-ui-plugins-demo` directory
	- Running `build.py` will build & start storybook automatically
 	- No need to run 4 separate commands to do so
 

📜 Documentation
 - Modified README.md to reflect this change and provide instructions on using `build.py`

andreaslordos committed:
build: automate build & launch of storybook

- Added `build.py` in the `/packages/superset-ui-plugins-demo` directory
	- Running `build.py` will build & start storybook automatically
	- No need to run 4 separate commands to do so

- Modified README.md to reflect this change and provide instructions on
using `build.py`

ghost commented:
There were the following issues with this Pull Request

* Commit: f3dc4384aa1017381e684933e1000ab362e1450a
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

etr2460 reviewed (changes requested):
cool idea! Although I think the more standard way would be to add a command in the demo `package.json` file. Then you could just use bash as opposed to python to do `cd ../../ && yarn build && cd - && yarn storybook`

kristw commented:
You also only need to build once. The subsequent storybook calls should not need to rebuild

andreaslordos commented:
> You also only need to build once. The subsequent storybook calls should not need to rebuild

@kristw I believe if you edit the code of a plugin/add your own, you will need to build Storybook again for the changes to take effect.

kristw commented:
@andreaslordos You can deep import from `@superset-ui/xxx/src/yyy` instead of `@superset-ui/xxx` (which will point to the built files) during development.

andreaslordos committed:
build: delete build.py

Get rid of build.py and transfer commands for
automated building & launching to package.json

ghost commented:
There were the following issues with this Pull Request

* Commit: f3dc4384aa1017381e684933e1000ab362e1450a
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 45123495022c542b8e5d085f34d3ae792b405f67
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

andreaslordos committed:
build: automate build & launch of storybook

Use "sudo npm run build" to build & launch Storybook

ghost commented:
There were the following issues with this Pull Request

* Commit: f3dc4384aa1017381e684933e1000ab362e1450a
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 45123495022c542b8e5d085f34d3ae792b405f67
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 2e1af85bd245b3f1001296724b7854b1cce78777
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

andreaslordos committed:
docs: update readme

Reflect changes in automated building & launching
of Storybook, now uses command "npm run build"
instead of "python build.py"

ghost commented:
There were the following issues with this Pull Request

* Commit: f3dc4384aa1017381e684933e1000ab362e1450a
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 45123495022c542b8e5d085f34d3ae792b405f67
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 2e1af85bd245b3f1001296724b7854b1cce78777
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 8bd9cc013b6c3b53f71c6a5b6ac2ffc9a96fc685
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

andreaslordos commented:
> cool idea! Although I think the more standard way would be to add a command in the demo `package.json` file. Then you could just use bash as opposed to python to do `cd ../../ && yarn build && cd - && yarn storybook`

@etr2460 thank you for the suggestion! I have modified my PR accordingly

etr2460 commented on packages/superset-ui-plugins-demo/package.json:
```diff
@@ -9,7 +9,8 @@
     "deploy-demo": "npm run demo:clean && npm run demo:build && npm run demo:publish && npm run demo:clean",
-    "storybook": "start-storybook -p 9001"
+    "storybook": "start-storybook -p 9001",
+    "build": "cd ../../ && yarn build && cd - && yarn storybook"
```
Much better! Maybe one final nit, I'd name this something like `build-and-run` or maybe follow the example with `demo`, naming the old `storybook` command `storybook:run` and this one just `storybook`

andreaslordos replied:
Addressed commit lint issues (I think) & renamed the script name to "storybook" - thank you for the suggestions!

etr2460 reviewed (commented):
Make sure you address the commit lint issues too (format the commit title properly)

andreaslordos committed:
build: rename npm script

rename npm script from "build" to "storybook" and update readme.md to reflect change

andreaslordos committed:
docs: amend error in docs

documentation updated to reflect changes in package.json - command to run story

etr2460 commented on README.md:
````diff
@@ -106,7 +106,13 @@ You can demo your changes by running the storybook demo locally with the followi
+Alternatively, you can demo your changes by using the following command while in `packages/superset-ui-plugins-demo`:
+
+```sh
+npm run storybook
````
one final nit, this should be `yarn storybook`

andreaslordos replied:
@etr2460 fixed

etr2460 reviewed (changes requested):
almost there!

andreaslordos committed:
docs: change 'npm run' to 'yarn'

replace 'npm run storybook' with 'yarn storybook'

andreaslordos commented:
Requesting review @etr2460

etr2460 reviewed (approved):
lgtm, thanks for the contribution!
